### ISSUE TITLE
Access Proxy Id through interface triggers Lazy Load

### ISSUE THREAD

**Issue by MagicShoebox** [nhibernate/nhibernate-core#2184]
Hi,

I've encountered an issue where accessing the Id property of a Proxy via an interface causes the Entity to be loaded.

```
public interface IEntity
{
    int Id { get; }
}

public class Entity : IEntity
{
    public virtual int EntityId { get; set; }

    int IEntity.Id => EntityId;
}

public void Test(ISessionFactory sessionFactory, int id)
{
    Entity entity;
    IEntity iEntity;
    using (ISession session = sessionFactory.OpenSession())
    {
        var x = session.Load<Entity>(id);
        entity = x;
        iEntity = x;
    }

    Assert.Equal(id, entity.EntityId);  // OK
    Assert.Equal(id, iEntity.Id);       // LazyInitializationException
}

```

I saw several threads that looked related, but unfortunately they dealt with more complicated situations (subclasses, etc.) and I'm not well-versed in NHibernate's internals. Most seem to indicate issues had been fixed, though. My issue is showing up on 5.2.5.

Is there something I am not understanding, or a workaround available?

Thanks!

**Comment by MagicShoebox** [comment]
Related issues:
- **https://github.com/nhibernate/nhibernate-core/pull/2066**
- https://github.com/nhibernate/nhibernate-core/issues/2085
- https://github.com/nhibernate/nhibernate-core/issues/2052

**Comment by fredericDelaporte** [comment]
Your case will not be fixed. You should remove the explicit implementation and only use the implicit ones. Then it should work with NHibernate 5.3, but it may depend on how we fix #2085, if we fix it.

With an explicit implementation of an interface, if the entity is mapped as the concrete class, there are no way NHibernate could tell if the Id property from the interface will actually match the Id from the mapped concrete class. So instead of doing some wild guess, it must trigger the proxy loading to simply call it and see what it yields.

**Comment by MagicShoebox** [comment]
Ah, that’s unfortunate but makes sense.

Thank you for the explanation!
